src/main/java/sirius/db/es/constraints/ElasticQueryCompiler.java

```diff
@@ -49,4 +52,22 @@ protected ElasticConstraint compileSearchToken(Mapping field, QueryField.Mode mo
 
         return Elastic.FILTERS.prefix(field, value.replace("*", "").toLowerCase());
     }
+
+    @Override
+    protected Tuple<Mapping, Property> resolvedNestedProperty(Property property, Mapping mapping, String nestedPath) {
+        if (property instanceof BaseMapProperty) {
+            return Tuple.create(Mapping.named(nestedPath), property);
+        }
+
+        return null;
+    }
+
+    @Override
+    protected ElasticConstraint compileFieldEquals(Mapping field, Property property, FieldValue value) {
+        if (value.getValue() instanceof String stringValue && (property instanceof BaseMapProperty)) {
```

Comment:
```suggestion
        if (value.getValue() instanceof String stringValue && property instanceof BaseMapProperty) {
```

Oder beides klammern?